atom/browser/api/atom_api_cookies.cc

```diff
@@ -0,0 +1,355 @@
+      is_secure != cookie.IsSecure())
+    return false;
+  if (filter->GetBoolean("session", &session) &&
+      session != cookie.IsPersistent())
+    return false;
+  return true;
+}
+
+}  // namespace
+
+namespace mate {
+
+template<>
+struct Converter<net::CanonicalCookie> {
+  static v8::Local<v8::Value> ToV8(v8::Isolate* isolate,
+      const net::CanonicalCookie& val) {
+    mate::Dictionary dict(isolate, v8::Object::New(isolate));
+    dict.Set("name", val.Name());
+    dict.Set("value", val.Value());
+    dict.Set("source", val.Source());
+    dict.Set("domain", val.Domain());
+    dict.Set("path", val.Path());
+    dict.Set("secure", val.IsSecure());
+    dict.Set("http_only", val.IsHttpOnly());
+    dict.Set("session", val.IsPersistent());
+    if (!val.IsPersistent())
+      dict.Set("expirationDate", val.ExpiryDate().ToDoubleT());
+    return dict.GetHandle();
+  }
+};
+
+}  // namespace mate
+
+namespace atom {
+
+namespace api {
+
+Cookies::Cookies() {
+}
+
+Cookies::~Cookies() {
+}
+
+void Cookies::Get(const base::DictionaryValue& options,
+                  const CookiesCallback& callback) {
+  // The filter will be deleted manually after callback function finishes.
+  base::DictionaryValue* filter = options.DeepCopyWithoutEmptyChildren();
+
+  content::BrowserThread::PostTask(BrowserThread::IO, FROM_HERE,
+      base::Bind(&Cookies::GetCookiesOnIOThread, base::Unretained(this),
+        filter, callback));
+}
+
+void Cookies::GetCookiesOnIOThread(const base::DictionaryValue* filter,
+                                   const CookiesCallback& callback) {
+  net::CookieStore* cookie_store =
+      AtomBrowserContext::Get()->url_request_context_getter()
+      ->GetURLRequestContext()->cookie_store();
+  std::string url;
+  filter->GetString("url", &url);
+  if (!GetCookieListFromStore(cookie_store, url,
+          base::Bind(&Cookies::OnGetCookies, base::Unretained(this), filter,
+              callback))) {
+    BrowserThread::PostTask(BrowserThread::UI, FROM_HERE,
+      base::Bind(&RunGetCookiesCallbackOnUIThread, filter, "Url is not valid",
+        net::CookieList(), callback));
+  }
+}
+
+void Cookies::OnGetCookies(const base::DictionaryValue* filter,
+                           const CookiesCallback& callback,
+                           const net::CookieList& cookie_list) {
+  net::CookieList result;
+  for (const auto& cookie : cookie_list) {
+    if (MatchesCookie(filter, cookie))
+      result.push_back(cookie);
+  }
+  BrowserThread::PostTask(BrowserThread::UI, FROM_HERE, base::Bind(
+        &RunGetCookiesCallbackOnUIThread, filter, "", result, callback));
+}
+
+void Cookies::Remove(const base::DictionaryValue& details,
+                     const CookiesCallback& callback) {
+  std::string url, name;
+  std::string error_message;
+  if (!details.GetString("url", &url) || !details.GetString("name", &name)) {
+    error_message = "Details(url, name) of removing cookie are required.";
+  }
+  GURL gurl(url);
```

Comment:
There is a `mate::Converter` for `GURL` so you don't have to construct a `gurl` here.